library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -994,6 +1001,9 @@ public int getMaxRetryCount()
    */
   public void setMaxRetryCount(int maxRetryCount)
   {
+    if (maxRetryCount < 0) {
+      throw new IllegalArgumentException("maxRetryCount should be greater than or equal to 0.");
+    }
```

Comment:
@chaithu14 agreed. Will raise a ticket for the same on APEXCORE. Can the existing changes be merged? I've added the @ Min constraint annotation so that it can be picked up.